src/sentry_options.c

```diff
@@ -276,6 +276,29 @@ sentry_options_get_http_proxy(const sentry_options_t *opts)
     return sentry_options_get_proxy(opts);
 }
 
+void
+sentry_options_set_read_proxy_from_environment(sentry_options_t *opts, int val)
+{
+    opts->read_proxy_from_environment = !!val;
+}
+
+int
+sentry_options_get_read_proxy_from_environment(const sentry_options_t *opts)
+{
+    return opts->read_proxy_from_environment;
+}
+
+void
+sentry__set_proxy_from_environment(sentry_options_t *opts)
+{
+    const char *https_proxy = getenv("https_proxy");
+    if (https_proxy) {
+        sentry_options_set_proxy(opts, https_proxy);
+    } else {
+        sentry_options_set_proxy(opts, getenv("http_proxy"));
```

Comment:
What happens if the env contains `http_proxy=http://user:pass@my-proxy:12345/`? Do we support this everywhere? Shouldn't we parse the URL coming from the env?

Reply:
Afaict this seems to work, but it makes sense to add tests that add `--proxyauth=user:pass` to the mitmproxy run command to make sure this is the case.

Reply:
If it works on all configs and thus doesn't require us to clean URLs, we should have tests to back that claim.